app/lib/scorecard/backend.dart

```diff
@@ -218,23 +219,68 @@ class ScoreCardBackend {
         if (size > _reportSizeDropThreshold) {
           _logger.reportError(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
-          // Size trimming is temporarily disabled.
-          // TODO: re-enable after fixing the underlying issues with the verbose reports and package:appengine.
-          return false;
+          return true;
         } else if (size > _reportSizeWarnThreshold) {
           _logger.warning(
               '$reportType report exceeded size threshold ($size > $_reportSizeWarnThreshold)');
         }
         return false;
       }
 
-      if (sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
-        // TODO: replace with something meaningful
-        scoreCard.panaReportJsonGz = <int>[];
+      if (panaReport != null &&
+          sizeCheck(ReportType.pana, scoreCard.panaReportJsonGz)) {
+        scoreCard.updateReports(
+          panaReport: PanaReport(
+            timestamp: DateTime.now().toUtc(),
+            panaRuntimeInfo: panaReport.panaRuntimeInfo,
+            reportStatus: panaReport.reportStatus,
+            derivedTags: panaReport.derivedTags?.take(100).toList(),
+            allDependencies: panaReport.allDependencies?.take(100).toList(),
+            licenseFile: panaReport.licenseFile,
+            report: pana.Report(
+              sections: panaReport.report?.sections
+                      .map(
+                        (s) => s.summary.length < 1000
+                            ? s
+                            : pana.ReportSection(
+                                id: s.id,
+                                title: s.title,
+                                status: s.status,
+                                grantedPoints: s.grantedPoints,
+                                maxPoints: s.maxPoints,
+                                summary: [
+                                  s.summary.substring(0, 900),
+                                  '`Report exceeded size limit.`',
+                                ].join('[...]\n\n'),
+                              ),
+                      )
+                      .toList() ??
+                  [],
+            ),
+            flags: panaReport.flags,
+            urlProblems: panaReport.urlProblems?.take(10).toList(),
+          ),
+        );
       }
-      if (sizeCheck(ReportType.dartdoc, scoreCard.dartdocReportJsonGz)) {
-        // TODO: replace with something meaningful
-        scoreCard.dartdocReportJsonGz = <int>[];
+      if (dartdocReport != null &&
+          sizeCheck(ReportType.dartdoc, scoreCard.dartdocReportJsonGz)) {
```

Comment:
Nit: Could we rename `sizeCheck` to something indicating what a true result means? Perhaps something like `reportIsTooBig`